src/components/NWCBudgetEditor.tsx

```diff
@@ -0,0 +1,190 @@
+import {
+    createForm,
+    getValue,
+    required,
+    setValue,
+    SubmitHandler
+} from "@modular-forms/solid";
+import { NwcProfile } from "@mutinywallet/mutiny-wasm";
+import { For, Show } from "solid-js";
+
+import {
+    AmountEditable,
+    Button,
+    Checkbox,
+    InfoBox,
+    KeyValue,
+    TextField,
+    TinyText,
+    VStack
+} from "~/components";
+import { useI18n } from "~/i18n/context";
+
+export type BudgetForm = {
+    connection_name: string;
+    auto_approve: boolean;
+    budget_amount: string; // modular forms doesn't like bigint
+    interval: "Day" | "Week" | "Month" | "Year";
+};
+
+export function NWCBudgetEditor(props: {
+    initialProfile?: NwcProfile;
+    initialName?: string;
+    onSave: (value: BudgetForm) => Promise<void>;
+}) {
+    const i18n = useI18n();
+
+    const [budgetForm, { Form, Field }] = createForm<BudgetForm>({
+        initialValues: {
+            connection_name:
+                props.initialProfile?.name || props.initialName || "",
+            // If there's an initial profile, look at that, otherwise default to false
+            auto_approve: props.initialProfile
+                ? !props.initialProfile.require_approval
+                : false,
+            budget_amount:
+                props.initialProfile?.budget_amount?.toString() || "0",
+            interval: "Day"
```

Comment:
Last thing and then its perfect! This allows the interval to have the lastSelected value be displayed instead of only Day
``` suggestion
            interval: props.initialValues?.budget_period as BudgetForm["interval"] || "Day"
```

Reply:
nice thank you